__tests__/app.test.js

```diff
@@ -311,3 +311,118 @@ describe("GET /api/users", () => {
+      .get("/api/articles?sort_by=topic")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeSortedBy("topic", { descending: true });
+      });
+  });
+  it("should get all articles sorted by author", () => {
+    return request(app)
+      .get("/api/articles?sort_by=author")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeSortedBy("author", { descending: true });
+      });
+  });
+  it("should get all articles sorted by created_at", () => {
+    return request(app)
+      .get("/api/articles?sort_by=created_at")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeSortedBy("created_at", { descending: true });
+      });
+  });
+  it("should get all articles sorted by votes", () => {
+    return request(app)
+      .get("/api/articles?sort_by=votes")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeSortedBy("votes", { descending: true });
+      });
+  });
+  it("should get all articles sorted by article_img_url", () => {
+    return request(app)
+      .get("/api/articles?sort_by=article_img_url")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeSortedBy("article_img_url", { descending: true });
+      });
+  });
+  it("should get all articles in asc order", () => {
+    return request(app)
+      .get("/api/articles?order=asc")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeSorted({ descending: false });
+      });
+  });
+  it("should get all articles sorted by title in ascending order", () => {
+    return request(app)
+      .get("/api/articles?sort_by=title&order=asc")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeSortedBy("title", { descending: false });
+      });
+  });
+  it("400 bad request if attempt to sort by body", () => {
+    return request(app)
+      .get("/api/articles?sort_by=body")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad request");
+      });
+  });
+  it("400 bad request if attempt to sort by a property that is not allowed", () => {
+    return request(app)
+      .get("/api/articles?sort_by=potatoes")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad request");
+      });
+  });
+  it("400 bad request if attempt to order with a parameter that is not allowed", () => {
+    return request(app)
+      .get("/api/articles?order=kennedy")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad request");
+      });
+  });
+  it("400 bad request if both order and sort parameters are not allowed", () => {
+    return request(app)
+      .get("/api/articles?order=kennedy&sort_by=duck")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad request");
+      });
+  });
```

Comment:
This is a really good test, but instead of "parameter" I would recommend renaming this to "value" instead as it isnt a parameter in this case but a query and the behaviour we are testing is based off of its value